### PR TITLE
Add docker parameter validation

### PR DESCRIPTION
**Pull request by pschorf**
## Changes proposed in this PR
- Add support for whitelisting docker parameters

## Why are we making these changes?
Not all docker parameters will be supported by kuberenetes, so adding an explicit whitelist

### CODE REVIEW

File: integration/tests/cook/test_basic.py [hunk 1]
```diff
@@ -2634,3 +2634,18 @@ def test_kubernetes_disallowed_volumes(self):
...
+    def test_disallowed_docker_parameters(self):
+        settings = util.settings(self.cook_url)
+        docker_parameters_allowed = util.get_in(settings, 'task-constraints', 'docker-parameters-allowed')
+        if docker_parameters_allowed is None:
+            self.skipTest('Requires docker-parameters-allowed')
+        docker_image = util.docker_image()
+        container = {'type': 'docker',
+                     'docker': {'image': docker_image,
+                                'parameters': [{'key': 'this_should_not_be_allowed',
+                                                'value': 'its_not_even_a_real_parameter'}]}}
```
Comment (dposada) [inline integration/tests/cook/test_basic.py]: ❤️

File: scheduler/test/cook/test/rest/api.clj [hunk 2]
```diff
@@ -316,7 +316,52 @@
...
+                                                                                     "value" "bar"}]}})]
+          (is (= 201
+                 (:status (h {:request-method :post
+                              :scheme :http
+                              :uri "/rawscheduler"
+                              :headers {"Content-Type" "application/json"}
+                              :authorization/user "test"
+                              :body-params {"jobs" [container-job]}}))))))
+
+      (testing "disallow docker paramters"
```
Comment (dposada) [inline scheduler/test/cook/test/rest/api.clj]:
```suggestion
      (testing "disallow docker parameters"
```
(or `params` as above)

File: scheduler/test/cook/test/rest/api.clj [hunk 3]
```diff
@@ -316,7 +316,52 @@
...
+                                                                                       "value" "bar"}]}})]
+            (is (= 400
+                   (:status (h {:request-method :post
+                                :scheme :http
+                                :uri "/rawscheduler"
+                                :headers {"Content-Type" "application/json"}
+                                :authorization/user "test"
+                                :body-params {"jobs" [container-job]}})))))))
+
+      (testing "allowed docker paramters"
```
Comment (dposada) [inline scheduler/test/cook/test/rest/api.clj]: Same as above

File: scheduler/test/cook/test/rest/api.clj (hunk 2 above)
Comment (pschorf) [reply scheduler/test/cook/test/rest/api.clj]: Done

File: scheduler/test/cook/test/rest/api.clj (hunk 3 above)
Comment (pschorf) [reply scheduler/test/cook/test/rest/api.clj]: Done

**Pull request merged**